In [5]:
import pandas as pd
import matplotlib.pyplot as plt
import numpy as np 

In [6]:
d = pd.read_csv("./data_assignment2.csv")
d

ID  Living_area  Rooms  Land_size  Biarea  Age  Selling_price
0    1          104    5.0      271.0    25.0   33        4600000
1    2           99    5.0     1506.0     6.0   88        4450000
2    3          133    6.0      486.0     NaN   44        4900000
3    4          175    7.0      728.0     NaN   14        6625000
4    5          118    6.0     1506.0     NaN   29        4600000
5    6          133    6.0      823.0     NaN   12        6650000
6    7           70    4.0     1685.0    70.0   53        2800000
7    8          134    6.0     1593.0    49.0   57        6100000
8    9           70    5.0     1120.0   102.0   81        3000000
9   10          133    6.0     1503.0     NaN   51        3000000
10  11          121    4.0     1575.0   112.0   81        4000000
11  12          136    6.0      381.0     NaN   42        4350000
12  13           86    4.0     1529.0    30.0   90        3025000
13  14          135    6.0      334.0     8.0   45        5215000
14  15          130    5.0     2095.0     NaN   63        4275000
15  16          104    5.0      399.0    13.0   59        4100000
16  17           66    2.0     1655.0    20.0   90        3100000
17  18          129    5.0      414.0    21.0   32        5370000
18  19           75    NaN     1698.0     NaN   75        3000000
19  20          160    6.0     2104.0     NaN   63        4175000
20  21           87    4.0      268.0     NaN   22        3850000
21  22          182    7.0     1851.0    50.0    4        6300000
22  23          135    5.0      224.0     7.0    3        5350000
23  24          100    4.0     1213.0    84.0   65        4370000
24  25          202    6.0     1253.0     NaN   54        4250000
25  26          136    5.0     1381.0     8.0   21        4800000
26  27          185    7.0      746.0    31.0   44        5940000
27  28          153    6.0     3285.0    43.0   96        6600000
28  29          135    5.0      223.0     NaN    2        5200000
29  30           73    4.0     1944.0    70.0   53        3925000
30  31          120    4.0     1679.0    10.0   51        4350000
31  32          122    5.0      567.0     NaN    7        5825000
32  33           63    NaN     1665.0    60.0   39        2500000
33  34          150    5.0     1335.0     NaN    8        6050000
34  35          210    6.0     1978.0     NaN    5        6100000
35  36          108    5.0      299.0     NaN   32        4800000
36  37          111    4.0      367.0     NaN   59        4200000
37  38           72    4.0     1645.0    50.0   62        3450000
38  39          170    7.0      626.0    30.0   40        6195000
39  40          142    6.0      700.0    22.0   32        6050000
40  41          170    7.0        NaN     NaN    0        1775000
41  42          120    4.0      891.0     NaN    6        5850000
42  43          107    4.0     1643.0     NaN   59        4960000
43  44          135    5.0      223.0     8.0    2        5200000
44  45          145    5.0     1363.0     NaN   13        6150000
45  46          168    6.0      656.0     7.0   81        2360000
46  47           69    3.0     1500.0    69.0   82        2700000
47  48          170    7.0     1098.0    20.0   51        5930000
48  49          160    6.0     1191.0     NaN    2        6800000
49  50          170    7.0     1637.0    25.0   11        4850000
50  51          158    6.0     1539.0    23.0   12        5000000
51  52          103    5.0      264.0     NaN   19        4245000
52  53          148    5.0      771.0    30.0   32        5725000
53  54          139    5.0      218.0     NaN    3        5275000
54  55          118    6.0      937.0   118.0   45        4850000
55  56          159    7.0     1315.0    30.0   64        4825000

**1a) Find a linear regression model that relates the living area to the selling price.**
**If you did any data cleaning step(s), describe what you did and explain why**

**1b) What are the values of the slope and intercept of the regression line?** 

**1c) Use this model to predict the selling prices of houses which have living area**
**100 m2, 150 m2 and 200 m2**

**1d) Draw a residual plot.** 

**2a) Use a confusion matrix to evaluate the use of logistic regression to classify the iris data set**

**2b) Use k-nearest neighbours to classify the iris data set with some different**
**values for k, and with uniform and distance-based weights. What will happen** 
**when k grows larger for the different cases? Why?** 

**2c) Compare the classification models for the iris data set that are generated by** 
**k-nearest neighbours (for the different settings from question b) and by**
**logistic regression. Calculate confusion matrices for these models and discuss** 
**the performance of the various models**